### ISSUE TITLE
Explosions can break blocks.

### ISSUE THREAD

**Issue by XakepSDK** [FoxDenStudio/FoxGuard#36]
I made  a region with flag "root false"
My friend can break blocks with dynamite. Then i tested this plugin with IC2, when IC2 tnt or nuke explodes, no blocks will break, but blocks will be dropped as drop.
Sorry for bad english.

**Comment by XakepSDK** [comment]
Forgot versions:
forge-1.8.9-11.15.1.1902-1.8.9
FoxGuard latest, FoxCore-server latest, ic2 latest, minecraft 1.8.9.

**Comment by gravityfox** [comment]
can't help you there.
This is a case where mods are doing things that they shouldn't be doing.
This is a IC2 issue, not a foxguard issue.

**Comment by XakepSDK** [comment]
And vanilla TNT? Vanilla TNT can break blocks.

**Comment by gravityfox** [comment]
lol you're right, and I think this is derpy Sponge code. Let me look into it.

**Comment by gravityfox** [comment]
yeah i think sponge isn't firing the correct events for explosions. This will need to wait on a fix from sponge's end, not my end.

**Comment by XakepSDK** [comment]
It's your fault. RedProtect works fine.

**Comment by gravityfox** [comment]
Wow... Have you forgotten that i'm the developer here?

Also i just took SpongeForge master branch and lit a tnt block
The only event that fires is ExplosionEvent.Detonate and it doesn't have the player cause.

This is a sponge bug.

you can see the issue here: https://github.com/SpongePowered/SpongeCommon/issues/833

**Comment by XakepSDK** [comment]
And i write paid spigot plugins. I'm developer too :D

**Comment by gravityfox** [comment]
See Foxguard has no problem STOPPING explosions. It just throws a fit when it can't figure out the player cause. The ExplosionEvent is missing a player cause as of right now, so FoxGuard just assumes a creeper blew up or something. This is by design, as it's very different when a player lights TNT vs when a creeper explodes. It also doesn't help that Sponge isn't firing an event for the blocks being broken, which it used to do.

Basically, it isn't my fault. The code never changed on my end.

If you're a developer then you should know that the number one way to piss off any developer is to blatantly say that something is their fault without stating a concrete argument about what they are doing wrong.

TL;DR If you come barging in all high and mighty like you know what's going on when you really don't, you can fuck off. Next time, either give me a solid argument with code snippets, or don't pretend you know what you're talking about.

**Comment by XakepSDK** [comment]
Why are you so aggressive?
I said, that ALL works fine in redprotect, you can just find a piece of code, or event, that uses redprotect, or you can piss on the tester. Thanks...
Codemonkey.

**Comment by gravityfox** [comment]
I'm being aggressive because you could have just left it as, "it doesn't work" and just accepted my response of "sponge isn't firing the correct events". No, you had to then make the accusation that it was my fault.

**Comment by gravityfox** [comment]
I didn't think I had to do this but apparently people feel the need to share the link to this issue on other websites to show others how "aggressive" I am. So allow me to make a note to all of those people coming here for the first time and reading all of this.

People don't seem to realize is that this place is an issue tracker. My issue tracker.

This is where I keep track of what I need to fix and what I need to improve.

If someone comes along and decides that they want to make accusations and irrelevant comments, they are wasting my time. I am not inclined to be very kind to people who waste my time.

I'm not being aggressive by telling people to shut up when they are on my issue tracker. When I tell people to shut up, implied or otherwise, it's because they aren't being helpful. I'm already aware of the issue, and giving me other crap isn't going to solve the problem. I'm not here to have a conversation, I'm here to get shit fixed. If you want a conversation, you can find me on IRC.

If you don't believe me, just look back over the conversation. I didn't start getting annoyed until I was accused directly for the problem. ("It's your fault.") At that point I already knew the issue, and I closed it because I knew it was not something I can fix in FoxGuard. 

There is an unspoken rule in issue trackers that after an issue has been closed, you don't make comments unless you can come up with an argument as to why the issue should be reopened. That means that those comments need to be on topic and on point.

That didn't happen. Instead I just got accusations and self righteous comments, and I'm not about to bow down to that. I'm not being paid to work on this plugin. I don't have a client. I'm working on this because I enjoy making things for the Minecraft community.

If after all this you still think I'm aggressive, just remember, this is only when people step out of line and piss me off. I normally give hugs and cookies to people, and I'm pretty chatty on IRC and Sponge Forums.

tl;dr
My issue tracker. My rules.
Pointing out that people are being annoying, implied or otherwise, is not being aggressive.
If you're here to make an issue, be brief, be descriptive, and don't give me fluff.
I will love you as long as you don't annoy me, and honestly, I'm not easy to piss off.
However accusing me of something without a rational reason is the fastest way to do so.
